tasks/commit_update.py

```diff
@@ -44,67 +45,79 @@ def run_impl(
-                    branch.authors.append(commit.author_id)
-
-                previous_branch_head = (
-                    db_session.query(Commit)
-                    .filter(Commit.repoid == repoid, Commit.commitid == branch.head)
+                if pull is None:
+                    pull = Pull(
+                        repoid=repoid,
+                        pullid=commit.pullid,
+                        author_id=commit.author_id,
+                        head=commit.commitid,
+                    )
+                    db_session.add(pull)
+                else:
+                    previous_pull_head = (
+                        db_session.query(Commit)
+                        .filter(Commit.repoid == repoid, Commit.commitid == pull.head)
+                        .first()
+                    )
+                    if (
+                        previous_pull_head is None
+                        or previous_pull_head.deleted == True
+                        or previous_pull_head.timestamp < commit.timestamp
+                    ):
+                        pull.head = commit.commitid
+
+            if commit.branch is not None:
+                # upsert branch
+                branch = (
+                    db_session.query(Branch)
+                    .filter(Branch.repoid == repoid, Branch.branch == commit.branch)
                     .first()
                 )
 
-                if (
-                    previous_branch_head is None
-                    or previous_branch_head.deleted == True
-                    or previous_branch_head.timestamp < commit.timestamp
-                ):
-                    branch.head = commit.commitid
+                if branch is None:
+                    branch = Branch(
+                        repoid=repoid,
+                        branch=commit.branch,
+                        head=commit.commitid,
+                        authors=[commit.author_id],
+                    )
+                    db_session.add(branch)
+                else:
+                    if commit.author_id is not None:
+                        if branch.authors is None:
+                            branch.authors = [commit.author_id]
+                        elif commit.author_id not in branch.authors:
+                            branch.authors.append(commit.author_id)
+
+                    previous_branch_head = (
+                        db_session.query(Commit)
+                        .filter(Commit.repoid == repoid, Commit.commitid == branch.head)
+                        .first()
+                    )
+
+                    if (
+                        previous_branch_head is None
+                        or previous_branch_head.deleted == True
+                        or previous_branch_head.timestamp < commit.timestamp
+                    ):
+                        branch.head = commit.commitid
 
-            db_session.flush()
+                db_session.flush()
```

Comment:
This indent means only run `db_session.flush()` if `commit.branch is not None`, is it intended?

Reply:
honestly the db_session.flush() is a bit redundant here, it happens at the end of the task and after `run_impl` runs we commit anyways so it's safe to remove since it achieves nothing